### ISSUE TITLE
Language Server doesn't initialize correctly when opening single vue file with no workspace

### ISSUE THREAD

**Issue by vip32** [octref/vetur#160]
this error occurs as soon as i switch the demo file to type vue

[Error - 10:32:36 PM] Server initialization failed.
  Message: Request initialize failed with message: Cannot read property 'substr' of null
  Code: -32603

**Comment by octref** [comment]
Please provide vetur and vscode version and complete error log.

**Comment by vip32** [comment]
vscode 1.112
vetur 0.6.3

this is all from the output window:
[Error - 10:32:36 PM] Server initialization failed.
  Message: Request initialize failed with message: Cannot read property 'substr' of null
  Code: -32603

**Comment by octref** [comment]
Do you have any jsconfig or tsconfig files in your workspace?

**Comment by vip32** [comment]
not really, just opening a single vue file without a complete project.

**Comment by octref** [comment]
I can repro it thanks.

**Comment by octref** [comment]
The problem is `workspacePath` can be null when people open single file without workspace in VSCode in here:

https://github.com/octref/vetur/blob/master/server/src/modes/javascriptMode.ts#L55-L66

**Comment by octref** [comment]
@mjbvz Curious if you mind sharing VSCode's strategy for consuming TS's LS for single files? Seems the whole TS LS is built based on the assumption of a `workspacePath`.

**Comment by mjbvz** [comment]
We support the single file case and don't do very much special on our side. When we first tell TypeScript Server about a file, it tries to walk up the file's path to find a `tsconfig` or `jsconfig` that owns the file. Behind the scenes, I believe it then loads the entire project. This is why workspace symbol search will return results outside of the single file that is actually open

We really don't use workspace.rootPath in very many places

**Comment by Youdaman** [comment]
I'm also having this issue each time I open a .vue file -- is there any fix/workaround?

I don't mind clicking the close icon on the console that pops up but it'd be nice to not have to :)

**Comment by Youdaman** [comment]
Updated to the latest VSCode and vetur and still get:

```
vetur initialized
[Error - 12:17:22 pm] Server initialization failed.
  Message: Request initialize failed with message: Cannot read property 'substr' of null
  Code: -32603 ```

**Comment by alexsasharegan** [comment]
I have this issue regularly inside the `<script>` block with es6 import statements:
![vetur-language-server-error](https://cloud.githubusercontent.com/assets/16269252/26378864/32074416-3fcb-11e7-9960-aa8210469d29.gif)

**Comment by alexsasharegan** [comment]
Looks like my specific issue was a duplicate of #215 — which was solved with the release of v0.6.11
